### ISSUE TITLE
Castle.configure throws NullPointerException on 3.0.4

### ISSUE THREAD

**Issue by Tylos** [castle/castle-android#64]
Our error management tool has started showing the following crash in production. 

```
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Number.intValue()' on a null object reference
        at io.castle.highwind.android.f.a(f:64)
        at io.castle.highwind.android.p.f(p:7)
        at io.castle.highwind.android.Highwind.token(Unknown:2)
        at io.castle.android.Castle.id(Castle.java:109)
        at io.castle.android.Castle.createRequestToken(Castle.java:280)
        at io.castle.android.api.model.Event.<init>(Event.java:38)
        at io.castle.android.api.model.CustomEvent.<init>(CustomEvent.java:32)
        at io.castle.android.api.model.CustomEvent.<init>(CustomEvent.java:23)
        at io.castle.android.Castle.custom(Castle.java:181)
        at io.castle.android.Castle.registerLifeCycleCallbacks(Castle.java:99)
        at io.castle.android.Castle.configure(Castle.java:135)
        at ynab.app.application.App.onCreate(App.kt:213)
```
We have made no changes to how Castle is integrated into our codebase other than bumping to version `3.0.4`. The change log indicates this version only contains internal enhancements and stability improvements. As the crash has only started showing up after bumping Castle to the latest release we think the bug could have been introduced in version `3.0.4`. 

Unfortunately, the crash source is not public so we can't provide more information on why this could happen. We also have been unable to reproduce the crash. At the moment it is affecting a low percentage of users only (1 out 1000).

**Comment by brissmyr** [comment]
@Tylos Thanks for reporting this here. Sebastian who's the maintainer of the Android SDK is out this week and a few days into the next, but we'll make it high priority for him to look into into it the day he's back.

**Comment by Tylos** [comment]
Sounds good, thanks for the heads up @brissmyr!

**Comment by sebastiansimson** [comment]
I'm looking into this to see what is causing the issue for you and wondering if it would be possible for you to provide some more information

- What version where you on of the SDK before upgrading to 3.0.4?
- Could I get a list of affected devices and OS versions? Does not need to be a complete list, just a subset.

**Comment by Tylos** [comment]
> What version where you on of the SDK before upgrading to 3.0.4?

We were on version `3.0.3`

> Could I get a list of affected devices and OS versions? Does not need to be a complete list, just a subset.

We have crashes reports from 4 different users:
* All running on Android 12
* Three devices are Pixel 6, and one is a Pixel 6 Pro

One thing to note regarding the low number of users affected. We detected the issue early during our beta process, and as we had no way to know how the issue could scale we decided to roll back to version `3.0.3` before publishing the final version to production. Since rolling back, we have not seen any additional crashes.
